utils/raft.go

```
@@ -57,181 +76,1347 @@ type VoteResponse struct {
+
+	// Create a waitgroup to track completion
+	var wg sync.WaitGroup
+	responses := make(chan bool, totalPeers)
+
 	for _, peer := range rn.peers {
+		wg.Add(1)
 		go func(peer string) {
-			conn, err := grpc.Dial("localhost:"+peer, grpc.WithInsecure())
+			defer wg.Done()
+
+			conn, err := grpc.Dial(peer, grpc.WithInsecure())
```

golangci-lint (1.64.8)

SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)